src/clib-uninstall.c

```diff
@@ -0,0 +1,212 @@
+//
+// clib-uninstall.c
+//
+// Copyright (c) 2014 Stephen Mathieson
+// Copyright (c) 2021 clib authors
+//
+// MIT licensed
+//
+
+#include "asprintf/asprintf.h"
+#include "commander/commander.h"
+#include "http-get/http-get.h"
+#include "logger/logger.h"
+#include "parse-repo/parse-repo.h"
+#include "debug/debug.h"
+#include "parson/parson.h"
+#include "version.h"
+#include <stdlib.h>
+#include <string.h>
+
+#define CLIB_UNINSTALL_DEFAULT_TARGET "make uninstall"
+
+#if defined(_WIN32) || defined(WIN32) || defined(__MINGW32__) ||               \
+    defined(__MINGW64__) || defined(__CYGWIN__)
+#define setenv(k, v, _) _putenv_s(k, v)
+#endif
+
+debug_t debugger;
+
+static void setopt_prefix(command_t *self) {
+  setenv("PREFIX", (char *)self->arg, 1);
+  debug(&debugger, "set prefix: %s", (char *)self->arg);
+}
+
+static char *get_tarball_url(const char *owner, const char *name,
+                             const char *version) {
+  char *tarball = NULL;
+  int size = 0;
+
+  size = asprintf(&tarball, "https://github.com/%s/%s/archive/%s.tar.gz", owner,
+                  name, version);
+
+  if (-1 == size)
+    return NULL;
+  return tarball;
+}
+
+static char *get_tar_filepath(const char *name, const char *version) {
+  char *file = NULL;
+  int size = asprintf(&file, "%s-%s.tar.gz", name, version);
+  if (-1 == size)
+    return NULL;
+  return file;
+}
+
+static char *get_tmp_tarball(const char *file) {
+  char *tmp = NULL;
+  int size = asprintf(&tmp, "/tmp/%s", file);
+  if (-1 == size)
+    return NULL;
+  return tmp;
+}
+
+static char *get_untar_command(const char *file) {
+  char *cmd = NULL;
+  int size = 0;
+  size = asprintf(&cmd, "cd /tmp && tar -xf %s", file);
+  if (-1 == size)
+    return NULL;
+  return cmd;
+}
+
+static char *get_uninstall_target(const char *name, const char *version) {
+  int size = 0;
+  char *target = NULL;
+  char *dir = NULL;
+  char *pkg = NULL;
+  const char *val = NULL;
+  JSON_Value *root = NULL;
+  JSON_Object *obj = NULL;
+
+  size = asprintf(&dir, "/tmp/%s-%s", name, version);
+  if (-1 == size)
+    return NULL;
+
+  size = asprintf(&pkg, "%s/package.json", dir);
```

Comment:
I don't think we should merge this as-is. All of the existing commands support `clib.json`. For good UX, we should keep things consistent between commands.

IMO we have two options:
1. (easy) just replace `package.json` with `clib.json` and mention that `clib uninstall` doesn't support the old manifest name. This isn't the preferred solution, but could be done quickly, enabling us to move on to other features.
2. Update this to _first_ check for `clib.json` and fallback to the `package.json` manifest. This is what the other commands do, but is a bit more work.

WDYT?

Reply:
Yes, you are right! I completely overlooked this. I think the 2. option is more correct, as clib has this behavior elsewhere.